### ISSUE TITLE
Feature Request: Custom Script Parameters

### ISSUE THREAD

**Issue by Mobius5150** [Firebottle/Firebot#338]
It would be awesome if custom scripts could receive parameters based off of which button was pressed. The script could also expose a function that returns the set of parameters it supports, and their types.

E.g.
```javascript
exports.getRunParameters = function() {
    return new Promise((resolve, reject) => {
        resolve(
            {
                "param1": {
                    "type": "string",
                    "default": "someValue"
                },
                "param2": {
                    "type": "number",
                    "default": 1
                },
                "param3": {
                    "type": "enum",
                    "options": [ "A", "B", "C", "D" ],
                    "default": "A"
                }
            }
        );
    });
}

exports.run = function (buttonId, username, modules, parameters) {
    // parameters argument has an object with the parameters for the button, e.g.
    // parameters === {
    //     "param1": "someValue",
    //     "param2": 1,
    //     "param3": "A"
    // }
}
```

It might not be strictly necessary to return a promise from `getRunParameters`, but some scripts (like what I'm working on) might work better if they can connect to whatever they're interfacing with and retrieve a list of options.

**Comment by ebiggz** [comment]
This is now dev complete. I will write some documentation on what has changed before we release.

**Comment by Firebottle** [comment]
Released in the dank update 4.2.0

**Comment by ebiggz** [comment]
@Mobius5150 
Documentation has been updated for this
https://github.com/Firebottle/Firebot/wiki/Writing-Custom-Scripts
